packages/react-components/react-utilities/src/hooks/useMergedRefs.ts

```diff
@@ -4,7 +4,7 @@ import * as React from 'react';
  * A Ref function which can be treated like a ref object in that it has an attached
  * current property, which will be updated as the ref is evaluated.
  */
-export type RefObjectFunction<T> = React.RefObject<T> & ((value: T) => void);
+export type RefObjectFunction<T> = React.RefObject<T> & ((value: T | null) => void);
```

Comment:
```suggestion
export type RefObjectFunction<T> = React.RefObject<T> & React.RefCallback<T>;
```

should not we just reuse `RefCallback` there?

Reply:
They're not exactly equivalent!

```ts
namespace React {
    // equivalent to react's RefCallback signature
    export type RefCallback<T> = {
        bivarianceHack(value: T | null): void
    }['bivarianceHack']
}

type RefCallback<T> = (value: T | null) => void


type UnionOfRefCallbacks = RefCallback<1> | RefCallback<2>
type UnionOfReactRefCallbacks = React.RefCallback<1> | React.RefCallback<2>

// as React.RefCallback is bivariant this is true ✅
type X = UnionOfRefCallbacks extends React.RefCallback<1 | 2> ? true : false // true

// as RefCallback is contravariant this is false ❌
type Y = UnionOfReactRefCallbacks extends RefCallback<1 | 2> ? true : false // false
```

**Playground Link:** [Provided](https://www.typescriptlang.org/play/?#code/FAOwhgtgpgzgDmAxlABAJSkgLig3sFQlAemJSgEcBXASwDcwAbKEHLAexQCdNEsByGOigAzAMJNGAIyQBrFDBoBzcFio8CRKAA847LmwCecVBnGSZiWQB4AKgD4UAXjyaiRKfTBcaYEMgAJOQAKBkYqKAAuFFsUAB8UECpGRgBKaLp2GgATNxQAXwBtfk8GHz9AuX4AXWB84GAsY1NRCRTLGwdnFFCmCOjYhKSU1OdHTJyGxuaUAFUQGnYQAHkRMzbpOSEXdYs5awBGRwTd9v2AJntpkzmFpdWMbFPNq23hbAA6Z47D4-e+L6tPZWayXBqkFBgISPAHfOQoGhCUreXysFBYAAWiIRQiwXAiKEAoOTXVAADW680WKzWQLOr3I2iwLGy0N4WEB5jpNgO8RQlxQAH50fjUNEREwYKgIXiIuCyFDhJyXvJsYglniwGVUWwsUJseLGJKUIAZchJKAAmhS7tSYVg4fSdEyQCzFRsfjyEvyhTLRSgDUaIf6oEA)


### TL,DR:

`(value: T | null) => void`  allows less scenarios than `React.RefCallback`, so if we would be introducing a bigger API surface if we used `React.RefCallback` (this is nit picking though)